youtube_dl/extractor/instagram.py

```diff
@@ -227,6 +227,112 @@ def get_count(key, kind):
         }
 
 
+def extract_graphql(extractor, data, _id, query_hash, query_vars, timeline_fn,
+                    gis_tmpl=None):
+    # A utility function which parses GraphQL queries containing videos and
+    # generates a playlist out of them.
+    #
+    # It is used by two different extractors which use two different queries.
+
+    def get_count(suffix):
+        return int_or_none(try_get(
+            node, lambda x: x['edge_media_' + suffix]['count']))
+
+    csrf_token = data['config']['csrf_token']
+    rhx_gis = data.get('rhx_gis') or '3c7ca9dcefcf966d11dacf1f151335e8'
+
+    cursor = ''
+    for page_num in itertools.count(1):
+        variables = {
+            'first': 12,
+            'after': cursor,
+        }
+        variables.update(query_vars)
+        variables = json.dumps(variables)
+
+        if gis_tmpl:
+            gis_tmpls = [gis_tmpl]
+        else:
+            gis_tmpls = [
+                '%s' % rhx_gis,
+                '',
+                '%s:%s' % (rhx_gis, csrf_token),
+                '%s:%s:%s' % (rhx_gis, csrf_token, std_headers['User-Agent']),
+            ]
+
+        for gis_tmpl in gis_tmpls:
+            try:
+                query_json = extractor._download_json(
+                    'https://www.instagram.com/graphql/query/', _id,
+                    'Downloading JSON page %d' % page_num, headers={
+                        'X-Requested-With': 'XMLHttpRequest',
+                        'X-Instagram-GIS': hashlib.md5(
+                            ('%s:%s' % (gis_tmpl, variables)).encode('utf-8')).hexdigest(),
+                    }, query={
+                        'query_hash': query_hash,
+                        'variables': variables,
+                    })
+                media = timeline_fn(query_json)
+                gis_tmpl = gis_tmpl
```

Comment:
`_gis_tmpl` broken.

Reply:
I didn't understand it was a persistent cache at first. Fixed and commented.